src/manifolds/EssentialManifold.jl

```diff
@@ -0,0 +1,516 @@
+"""
+struct EssentialManifold <: AbstractPowerManifold{ℝ,Rotations{3},NestedPowerRepresentation}
+    is_signed::Bool
+    manifold::Rotations{3}
+end
+
+EssentialManifold(is_signed::Bool=true) = EssentialManifold(is_signed, Rotations(3))
+
+@doc raw"""
+    check_manifold_point(M::EssentialManifold, p; kwargs...)
+
+Check whether the matrix is a valid point on the [`EssentialManifold`](@ref) `M`,
+i.e. a Tuple of two SO(3) matrices.
+"""
+function check_manifold_point(M::EssentialManifold, p; kwargs...)
+    if length(p) != 2
+        return DomainError(
+            length(p),
+            "The point $(p) does not lie on $M, since it does not contain exactly two elements.",
+        )
+    end
+    return check_manifold_point(
+        PowerManifold(M.manifold, NestedPowerRepresentation(), 2),
+        p;
+        kwargs...,
+    )
+end
+
+"""
+    check_tangent_vector(M::EssentialManifold, p, X; check_base_point = true, kwargs... )
+
+Check whether `X` is a tangent vector to manifold point `p` on the [`EssentialManifold`](@ref) `M`,
+i.e. `X` has to be a Tuple of two `3`-by-`3` skew-symmetric matrices. The optional parameter `check_base_point`
+indicates, whether to call [`check_manifold_point`](@ref)  for `p`.
+"""
+function check_tangent_vector(M::EssentialManifold, p, X; check_base_point=true, kwargs...)
+    if check_base_point
+        mpe = check_manifold_point(M, p; kwargs...)
+        mpe === nothing || return mpe
+    end
+    if length(X) != 2
+        return DomainError(
+            length(X),
+            "$(X) is not a tangent vector to the manifold $M, since it does not contain exactly two elements.",
+        )
+    end
+    return check_tangent_vector(
+        PowerManifold(M.manifold, NestedPowerRepresentation(), 2),
+        p,
+        X;
+        check_base_point=check_base_point,
+        kwargs...,
+    )
+end
+
+@doc raw"""
+    distance(M::EssentialManifold, p, q)
+
+Compute the Riemannian distance between the two points `p` and `q` on the [`EssentialManifold`](@ref). This is done by
+computing the distance of the equivalence classes $[p]$ and $[q]$ of the points
+$p=(R_{p_1},R_{p_2}), q=(R_{q_1},R_{q_2}) ∈ SO(3)^2$, respectively. Two points in $SO(3)^2$ are equivalent iff their
+corresponding essential matrices, given by
+````math
+E = R_1^T [e_z]_{×}R_2,
+````
+are equal (up to a sign flip). Using the logarithmic map, the distance is given by
+````math
+\text{dist}([p],[q]) = \| \text{log}_{[p]} [q] \| = \| \log_p (S_z(t_{\text{opt}})q) \|,
+````
+where $S_z ∈ H_z = \{(R_z(θ),R_z(θ))\colon θ \in [-π,π) \}$ in which $R_z(θ)$ is the rotation around the z axis with angle
+$θ$ and $t_{\text{opt}}$ is the minimizer of the cost function
+````math
+f(t) = f_1 + f_2, \quad f_i = \frac{1}{2} θ^2_i(t), \quad θ_i(t)=d(R_{p_i},R_z(t)R_{b_i}) \text{ for } i=1,2,
+````
+where $d(⋅,⋅)$ is the distance function in $SO(3)$[^TronDaniilidis2017].
+"""
+distance(M::EssentialManifold, p, q) = norm(M, p, log(M, p, q))
```

Comment:
That's just the default implementation. By the way, does `EssentialManifold` assume that the horizontal component of all tangent vectors is zero?

Reply:
Then this could be removed, too, I'll check that the tests still work then; 

No the horizontal component is not necessary zero, and that is also never assumed.

Reply:
Ah, this is actually more complicated, since with the power manifold, the default ist _first_ to go down to the elements and call their logs, so to avoid that and call the correct log here, we had to “reset” it from the Power default to the Manifold default.

Reply:
> No the horizontal component is not necessary zero, and that is also never assumed.

Right, it always confuses me when operations on a quotient manifold are implemented in a way where `inner` from the embedding just works.

Reply:
Sure, that would require to always store the points in a way that the horizontal component is zero; though having both projections (horizontal and vertical) would be nice, one should not assume aways having such a form.